libgalois/include/katana/TopologyGeneration.h

```diff
@@ -196,9 +197,11 @@ class KATANA_EXPORT PropertyGenerator {
 /// );
 template <typename... Args>
 KATANA_EXPORT Result<void>
-AddNodeProperties(katana::PropertyGraph* pg, Args&&... generators) {
+AddNodeProperties(
+    tsuba::TxnContext* txn_ctx, katana::PropertyGraph* pg,
```

Comment:
at the very least, can we put TxnContext towards the end? PropertyGraph*, TxnContext*, Args&&?

Reply:
@ddn0 I've been trying to put `TxnContext` at the end when it's possible. But a parameter pack needs to be put at the end afaik, isn't it? I remember I put `TxnContext` at the end here before, and got compilation errors.

Reply:
At the very least, can you make it `(PropertyGraph*, TxnContext*, Args&&...)`?

Reply:
My original thought was that something named `...Context` conventionally tends to be put as either first or last argument. But now I've changed all of the `TxnContext`-s to be last possible argument.